.github/workflows/ci.yml

```diff
@@ -28,3 +28,56 @@ jobs:
       - uses: pre-commit/action@v2.0.2
         with:
           extra_args: --source ${{ github.event.pull_request.base.sha || 'HEAD~1' }} --origin ${{ github.event.pull_request.head.sha || 'HEAD' }}
+
+  build-and-upload-artifacts:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build the typescript code
+        run: cd typescript && yarn && yarn build
+      - uses: actions/upload-artifact@main
+        with:
+          name: build artifacts
+          path: build/
+
+  build-to-npm:
+    if: github.ref == 'refs/heads/master'
+    runs-on: ubuntu-latest
+    needs:
+      - run-depcheck
+      - build-and-upload-artifacts
+      - run-pre-commits
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14.x'
+      - name: Configure NPM authentication
+        run: |
+          yarn config set npmAlwaysAuth true
+          yarn config set npmAuthToken ${{ secrets.NPM_TOKEN }}
+      - name: Publish to yarn/npm
+        run: cd typescript && yarn npm publish
```

Comment:
One thing that I'm not sure is that the `build-and-upload-artifacts` moves to the `typescript` folder, but not sure if this is reset when this action is run or if I should remove `cd typescript`